bokehjs/src/lib/models/annotations/toolbar_panel.ts

```diff
@@ -38,14 +41,26 @@ export class ToolbarPanelView extends AnnotationView {
     super.render()
   }
 
+  private _invalidate_toolbar = true
+  private _previous_bbox: BBox = new BBox()
+
   protected _render(): void {
-    this.el.style.position = "absolute"
-    this.el.style.overflow = "hidden"
+    // TODO: this should be handled by the layout
+    const {bbox} = this.panel
+    if (!this._previous_bbox.equals(bbox)) {
+      position(this.el, bbox)
+      this._previous_bbox = bbox
+    }
+
+    if (this._invalidate_toolbar) {
+      this.el.style.position = "absolute"
+      this.el.style.overflow = "hidden"
+      this._toolbar_view.render()
+      empty(this.el)
+      this.el.appendChild(this._toolbar_view.el)
+      this._invalidate_toolbar = false
```

Comment:
is `_invalidate_toolbar` only signaling the first render, or are there circumstances that could cause it to become true again?